### PR TITLE
Pimp the toolbar (for some values of "pimp")

### LINKED ISSUE #1328: COM reflection should pick up constant values
Instead of a `Declaration` for `DeclarationType.Constant` and `DeclarationType.EnumMember`, the code that inspects the types of COM libraries should generate a `ValuedDeclaration` to store the value.

The value could be displayed in the "current selection" status in the `RubberduckCommandBar`, and in other places such as the _Code Explorer_.

### PR DESCRIPTION
**Pull request by comintern**
Add default values to enumeration members, consts, and parameters, closes #1328

### CODE REVIEW

**Comment by comintern** [comment]
WIP pending unit tests (which VS managed to wipe out...).

File: Rubberduck.Parsing/Symbols/ParameterDeclaration.cs [hunk 1]
```diff
@@ -39,43 +41,51 @@ public class ParameterDeclaration : Declaration
...
             IsByRef = isByRef;
-            IsImplicitByRef = isByRef && ((VBAParser.ArgContext) context)?.BYREF() == null;
+            IsImplicitByRef = isByRef && argContext?.BYREF() == null;
             IsParamArray = isParamArray;
+
+            if (!isOptional || argContext?.argDefaultValue() == null)
+            {
+                return;
+            }
+            DefaultValue = argContext.GetDescendent<VBAParser.LiteralExprContext>()?.GetText() ?? string.Empty;
```
Comment (MDoerner) [inline Rubberduck.Parsing/Symbols/ParameterDeclaration.cs]: This does not look appropriate. Default members are not restricted to literals, just to compile time constants.

The following is completely legal:
```vba
Private Const bar As Long = 2

Private Enum Foo
   FooBar = bar + 15 - Module1.OtherModuleConst
End Enum


Private Sub DoSomething(Optional baz As Long = bar * 15 + Foo.FooBar + Module1.OtherModuleConst * FooBar)
   Debug.Print baz
End Sub
```
Here, `OtherModuleConst` is a public constant declared in `Module1`.

Until we can evaluate the expression context, I would propose to use `argContext.argDefaultValue().Expression.GetText()`.

Comment (comintern) [reply Rubberduck.Parsing/Symbols/ParameterDeclaration.cs]: Not sure why I was thinking it had to be a literal. This is fixed.

File: Rubberduck.Parsing/Symbols/DeclarationSymbolsListener.cs [hunk 2]
```diff
@@ -289,6 +289,22 @@ private void DeclareControlsAsMembers(IVBComponent form)
...
+                        new QualifiedMemberName(_qualifiedModuleName, identifierName),
+                        _parentDeclaration, 
+                        _currentScope, 
+                        asTypeName, 
+                        asTypeContext, 
+                        typeHint, 
+                        annotations,
+                        accessibility, 
+                        declarationType,
+                        context.GetDescendent<VBAParser.ExpressionContext>()?.GetText() ?? string.Empty,
```
[the remark this answers is not in the document]
Comment (comintern) [reply Rubberduck.Parsing/Symbols/DeclarationSymbolsListener.cs]: Good to know about the performance implications of `GetDescendent`. This is also fixed.

**Pull request merged**